src/components/pages/profile/file-input.tsx

```diff
@@ -0,0 +1,118 @@
+'use client'
+
+import ImageCard from '@/components/pages/auth/image-card'
+import { Icons } from '@/components/ui/icons'
+import { Progress } from '@/components/ui/progress'
+import { PROGRESS_TIMEOUT_DELAY } from '@/constants'
+import { cn } from '@/lib/utils'
+import Image from 'next/image'
+import { FC, FormEvent, useRef, useState } from 'react'
+
+type FileInputProps = {
+  className?: string
+}
+
+const FileInput: FC<FileInputProps> = ({ className }) => {
+  const fileInputRef = useRef<HTMLInputElement>(null)
+
+  const [_, setImage] = useState<File | undefined>()
+
+  const [preview, setPreview] = useState<string | undefined>()
+
+  const [progress, setProgress] = useState<number>(0)
+
+  const clearSelectFile = () => {
+    setPreview(undefined)
+  }
+
+  const handleSelectFile = () => {
+    fileInputRef.current?.click()
+  }
+
+  const handleImageLoad = (file: File) => {
+    setImage(file)
+
+    const fileReader = new FileReader()
+
+    fileReader.readAsDataURL(file)
+
+    fileReader.onloadstart = () => {
+      setProgress(0)
+    }
+    fileReader.onprogress = (progress) => {
+      setProgress((progress.loaded / progress.total) * 100)
+    }
+    fileReader.onload = () => {
+      setPreview(fileReader.result as string)
+      setTimeout(() => setProgress(0), PROGRESS_TIMEOUT_DELAY)
+    }
+  }
+
+  const handleImageChange = (event: FormEvent<HTMLInputElement>) => {
+    const target = event.target as HTMLElement & {
+      files: FileList
+    }
+
+    if (target.files.length > 0) {
+      handleImageLoad(target.files[0])
+    }
+  }
+
+  return (
+    <ImageCard
+      className={cn(
+        '-my-[4.375rem] flex-shrink-0 hidden sm:flex items-center justify-center py-32 px-[3.25rem]',
+        className
+      )}
+    >
+      <div className="relative">
+        {preview ? (
+          <div>
+            <Image
+              alt="avatar"
+              className="w-[144px] h-[144px] rounded-3xl"
+              src={preview}
+            />
```

Comment:
Such code causes too much errors in console (I don't wanna to explain it - you can see it yourself). I suggest such code which would cause exactly the same result as you shown on the video.

I also recommend you to check out how to deal with [Next.js Images](https://nextjs.org/docs/app/api-reference/components/image)

```suggestion
          <div className="relative w-[144px] h-[144px] overflow-hidden rounded-3xl">
            <Image
              alt="avatar"
              fill
              src={preview}
            />
```

Reply:
Thanks, I'll fix